internal/pkg/manifest/env.go

```diff
@@ -102,9 +109,98 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []SecurityGroupRule `yaml:"ingress,omitempty"`
+	Egress  []SecurityGroupRule `yaml:"egress,omitempty"`
+}
+
+func (cfg securityGroupConfig) isEmpty() bool {
+	return len(cfg.Ingress) == 0 && len(cfg.Egress) == 0
+}
+
+// SecurityGroupRule holds the security group ingress and egress configs
+type SecurityGroupRule struct {
+	CidrIP      string      `yaml:"cidr"`
+	PortsConfig PortsConfig `yaml:"ports"`
+	IpProtocol  string      `yaml:"ip_protocol"`
+}
+
+// Ports is a custom type which supports unmarshaling yaml which
+// can either be of type int or type PortsRangeBand.
+type PortsConfig struct {
+	Port  *int            // 0 is a valid value, so we want the default value to be nil.
+	Ports *PortsRangeBand // Mutually exclusive with port.
+}
+
+// PortsRangeBand is a number range with to and from port values.
+type PortsRangeBand string
+
+// Parse parses Ports string and returns the from and to port values.
+// For example: 1-100 returns	1 and 100.
+func (str PortsRangeBand) Parse() (fromPort int, toPort int, err error) {
+	ports := strings.Split(string(str), "-")
```

Comment:
should we check if the length is equal to 2? otherwise it's possible that "0-1-2" is a valid band